MIDAS/src/finite-state-machines/fsm.cpp

```diff
@@ -0,0 +1,343 @@
+#include "fsm.h"
+#include "thresholds.h"
+#include "systems.h"
+#include <cmath>
+
+// code for helper functions
+
+float FSM::getAcceleration(std::array<HighGData, 8> hg)
+{ 
+    float avg = 0;
+
+    for (size_t i = 0; i < hg.size(); i++)
+    {
+        avg += (hg.at(i).gz / (float)hg.size());
+    }
+
+    return avg;
+}
+
+float FSM::getAltitude(std::array<Barometer, 8> bar)
+{
+    float avg = 0;
+
+    for (size_t i = 0; i < bar.size(); i++)
+    {
+        avg += (bar.at(i).altitude / (float)bar.size());
+    }
+
+    return avg;
+}
+
+double FSM::getJerk(std::array<HighGData, 8> hg)
+{
+
+    double avg1 = 0;
+    for (size_t i = 0; i < hg.size() / 2; i++)
+    {
+        avg1 += (hg.at(i).gz / hg.size());
+    }
+
+    double avg2 = 0;
+    for (size_t i = hg.size() / 2; i < hg.size(); i++)
+    {
+        avg2 += (hg.at(i).gz / hg.size());
+    }
+
+    double sec = tick_to_ms(hg.size()) * 1000;
+
+    return (avg2 - avg1) / (sec);
+}
+
+double FSM::getVerticalSpeed(std::array<Barometer, 8> bar)
```

Comment:
getting vertical speed from such a small slice of barometer data is problematic. This could be as little as 8 ms of barometer data, not very reliable for getting speed